pipeline/config_test.go

```diff
@@ -0,0 +1,278 @@
+				ops = append(ops, newDummyJSON("json_parser"))
+				return ops
+			},
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser1"))
+				return ops
+			}(),
+		},
+		{
+			"multi_op_rename",
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+
+				return ops
+			},
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser1"))
+				ops = append(ops, newDummyJSON("json_parser2"))
+				ops = append(ops, newDummyJSON("json_parser3"))
+				ops = append(ops, newDummyJSON("json_parser4"))
+				return ops
+			}(),
+		},
+		{
+			"different_ops",
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyCopy("copy"))
+				ops = append(ops, newDummyCopy("copy"))
+
+				return ops
+			},
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser1"))
+				ops = append(ops, newDummyJSON("json_parser2"))
+				ops = append(ops, newDummyCopy("copy"))
+				ops = append(ops, newDummyCopy("copy1"))
+				return ops
+			}(),
+		},
+		{
+			"unordered",
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyCopy("copy"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyCopy("copy"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				return ops
+			},
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyCopy("copy"))
+				ops = append(ops, newDummyJSON("json_parser1"))
+				ops = append(ops, newDummyCopy("copy1"))
+				ops = append(ops, newDummyJSON("json_parser2"))
+				return ops
+			}(),
+		},
+		{
+			"already_renamed",
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser3"))
+				ops = append(ops, newDummyJSON("json_parser"))
+				return ops
+			},
+			func() Config {
+				var ops Config
+				ops = append(ops, newDummyJSON("json_parser"))
+				ops = append(ops, newDummyJSON("json_parser1"))
+				ops = append(ops, newDummyJSON("json_parser2"))
+				ops = append(ops, newDummyJSON("json_parser3"))
+				ops = append(ops, newDummyJSON("json_parser4"))
+				return ops
+			}(),
+		},
+		{
+			"iterate_twice",
```

Comment:
@djaglowski I added one already does an out of order iteration here.

Reply:
Fair enough. I would say that this test seems to cover more than the simplest form of this behavior, but it's fine.